File: Projeto_7/models/professor.py
# models/professor.py
class Professor:
    def __init__(self, nome, email, senha, especialidade):
        from models.autenticavel import UsuarioAutenticavel
        
        self.usuario_autenticavel = UsuarioAutenticavel(nome, email, senha)
        self.especialidade = especialidade
        self.disciplinas_ministradas = []  # Lista de disciplinas que o professor ensina

    def adicionar_disciplina(self, disciplina):
        """ Adiciona uma disciplina ao professor. """
        if disciplina not in self.disciplinas_ministradas:
            self.disciplinas_ministradas.append(disciplina)

File: projeto-8/Professor.cpp
#include "Professor.h"
#include <iostream>

Professor::Professor(std::string nome, std::string email, std::string senha, std::string areaDeAtuacao)
    : UsuarioAutenticavel(nome, email, TipoUsuario::PROFESSOR, senha), areaDeAtuacao(areaDeAtuacao) {
}

void Professor::adicionarDisciplina(const std::string& disciplina) {
    this->disciplinasMinistradas.push_back(disciplina);
}

bool Professor::autenticar(const std::string& senha) {
    return this->getSenha() == senha;
}

void Professor::gerarRelatorio() {
    std::cout << "--- Relatorio do Professor ---\n"
              << "Nome: " << this->nome << "\n"
              << "Area de Atuacao: " << this->areaDeAtuacao << "\n"
              << "------------------------------------\n"
              << "Disciplinas Ministradas:\n";

    if (disciplinasMinistradas.empty()) {
        std::cout << "Nenhuma disciplina ministrada.\n";
    } else {
        for (const auto& disciplina : disciplinasMinistradas) {
            std::cout << "  - " << disciplina << "\n";
        }
    }
}
